src/tools/pkcs11-tool.c

```diff
@@ -2733,8 +2736,49 @@ static void encrypt_data(CK_SLOT_ID slot, CK_SESSION_HANDLE session,
 	fprintf(stderr, "Using encrypt algorithm %s\n", p11_mechanism_to_name(opt_mechanism));
 	memset(&mech, 0, sizeof(mech));
 	mech.mechanism = opt_mechanism;
+	oaep_params.hashAlg = 0;
+
+	if (opt_hash_alg != 0 && opt_mechanism != CKM_RSA_PKCS_OAEP)
+		util_fatal("The hash-algorithm is applicable only to "
+			   "RSA-PKCS-OAEP mechanism");
 
+	/* set "default" MGF and hash algorithms. We can overwrite MGF later */
 	switch (opt_mechanism) {
+	case CKM_RSA_PKCS_OAEP:
```

Comment:
I see this is mostly copy-paste from the `decrypt_data()`. Is there a way to make this somehow share the code between these two function to avoid the need to fix possible issues in both places?